modules/models/behavior/behavior_model.hpp

```diff
@@ -73,19 +77,26 @@ class BehaviorModel : public modules::commons::BaseType {
     behavior_status_ = status;
   }
 
-  virtual Trajectory Plan(float delta_time,
+  virtual Trajectory Plan(float min_dt,
                           const world::ObservedWorld& observed_world) = 0;
 
-  virtual void ActionToBehavior(const Action& action) {};
 
   virtual std::shared_ptr<BehaviorModel> Clone() const {};
 
   Action GetLastAction() const { return last_action_; }
-  void SetLastAction(const Action action) { last_action_ = action; }
+  void SetLastAction(const Action& action) { last_action_ = action; }
+
+  // externally set action that can also be set from Python
+  Action GetAction() const { return action_to_behavior_; }
+  virtual void ActionToBehavior(const Action& action) {
+    action_to_behavior_ = action;
+  };
 
  private:
   dynamic::Trajectory last_trajectory_;
+  // can either be the last action or action to be executed
   Action last_action_;
+  Action action_to_behavior_;
```

Comment:
Do we need to support the action to behavior for all behavior models? I think it is sufficient for motion primitives or these types of behavior models

Reply:
Due to the fact that is is wrapped in the PyBehaviorModel I think there is no other option?